packages/hierarchies-react/src/presentation-hierarchies-react/itwinui/TreeErrorRenderer.tsx

```diff
@@ -3,78 +3,100 @@
  * See LICENSE.md in the project root for license terms and full copyright notice.
  *--------------------------------------------------------------------------------------------*/
 
-import { ComponentPropsWithoutRef } from "react";
+import { ComponentPropsWithoutRef, forwardRef, ForwardRefExoticComponent, RefAttributes } from "react";
 import { Anchor, Text, Tree } from "@itwin/itwinui-react/bricks";
 import { MAX_LIMIT_OVERRIDE } from "../internal/Utils.js";
 import { PresentationInfoNode } from "../TreeNode.js";
-import { useTree } from "../UseTree.js";
+import { HierarchyLevelDetails, useTree } from "../UseTree.js";
 import { FlatNode } from "./FlatTreeNode.js";
 import { useLocalizationContext } from "./LocalizationContext.js";
-import { TreeNodeRenderer } from "./TreeNodeRenderer.js";
+
+/** @alpha */
+export interface TreeErrorActionsProps {
+  /** A callback to reload a hierarchy level when an error occurs and `retry` button is clicked. */
+  reloadTree?: (options: { parentNodeId: string | undefined; state: "reset" }) => void;
+  /** Action to perform when the filter button is clicked for this node. */
+  onFilterClick?: (hierarchyLevelDetails: HierarchyLevelDetails) => void;
+}
 
 interface TreeErrorRendererOwnProps {
   node: FlatNode<PresentationInfoNode>;
 }
 
-type TreeErrorRendererProps = TreeErrorRendererOwnProps &
-  Pick<ComponentPropsWithoutRef<typeof TreeNodeRenderer>, "onFilterClick" | "reloadTree"> &
+type TreeErrorRendererProps = Pick<ComponentPropsWithoutRef<typeof Tree.Item>, "style"> &
+  TreeErrorRendererOwnProps &
+  TreeErrorActionsProps &
   Partial<Pick<ReturnType<typeof useTree>, "getHierarchyLevelDetails">>;
 
 /** @internal */
-export function TreeErrorRenderer({ node, getHierarchyLevelDetails, onFilterClick, reloadTree }: TreeErrorRendererProps) {
-  const { localizedStrings } = useLocalizationContext();
-  if (node.type === "ResultSetTooLarge") {
-    return (
-      <ResultSetTooLargeNode
-        limit={node.resultSetSizeLimit}
-        onOverrideLimit={getHierarchyLevelDetails ? (limit) => getHierarchyLevelDetails(node.parentNodeId)?.setSizeLimit(limit) : undefined}
-        onFilterClick={
-          onFilterClick
-            ? () => {
-                const hierarchyLevelDetails = getHierarchyLevelDetails?.(node.parentNodeId);
-                hierarchyLevelDetails && onFilterClick(hierarchyLevelDetails);
-              }
-            : undefined
-        }
-        aria-level={node.level}
-        aria-posinset={node.posInLevel}
-        aria-setsize={node.levelSize}
-      />
-    );
-  }
-
-  if (node.type === "NoFilterMatches") {
+export const TreeErrorRenderer: ForwardRefExoticComponent<TreeErrorRendererProps & RefAttributes<HTMLElement>> = forwardRef(
+  ({ node, getHierarchyLevelDetails, onFilterClick, reloadTree, ...rest }, forwardedRef) => {
```

Comment:
No need to have `...rest` if that contains only `style`.

Reply:
we just pass style along, for the future if more properties are going to be passed along I think rest is better as it does not clutter.